syntax = "proto3";

message Empty {

}


message LocationMessage {
  int32 id = 1;
  int32 person_id = 2;
  string coordinate = 3;
  string creation_time = 4;
  string wkt_shape = 5;
}

message LocationRequest {
    int32 id = 1;
}

service LocationService {
  rpc Retrieve(LocationRequest) returns (LocationMessage);
}


message PersonMessage {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message PersonMessageList {
  repeated PersonMessage persons = 1;
}

message PersonRequest {
    int32 id = 1;
}

service PersonService {
  rpc Create(PersonMessage) returns (PersonMessage);
  rpc Retrieve(PersonRequest) returns (PersonMessage);
  rpc RetrieveAll(Empty) returns (PersonMessageList);
}


message ConnectionMessage {
  LocationMessage location = 1;
  PersonMessage person = 2;  
}

message ConnectionsMessageList {
  repeated ConnectionMessage connections = 1;
}

message ConnectionRequest {
  string id = 1;
  string start_date = 2;
  string end_date = 3;
  string distance = 4;
}

service ConnectionService {
  rpc FindContacts(ConnectionRequest) returns (ConnectionsMessageList);
}